src/main/scala/rps/GameRepository.scala

```diff
@@ -12,15 +17,54 @@ trait GameRepository {
 
 class InMemoryGameRepository extends GameRepository {
   private val map = TrieMap.empty[UserId, Play]
+  private val db = Database.forConfig("h2mem1") // TODO: we should db.close this!
   private var nextInt = new AtomicInteger
 
+  val plays = TableQuery[Plays]
+  val setupAction = plays.schema.create
+  val setupFuture = db.run(setupAction)
+  Await.result(setupFuture, Duration.Inf)
+
   override def save(play: Play): UserId = {
-    val userId = UserId(nextInt.getAndIncrement)
-    map.put(userId, play)
-    userId
+    val id = nextInt.getAndIncrement()
+    val userMove = play.userMove
+    val computerMove = play.computerMove
+    val result = play.result
+    val plays = TableQuery[Plays]
+    val action = plays ++= Seq(
+      (id, userMove.toString(), computerMove.toString(), result.toString()),
+    )
+    val future = db.run(action)
+    Await.result(future, Duration.Inf) // assume this always succeeds
+    UserId(id)
   }
 
-  override def read(userId: UserId): Option[Play] =
-    map.get(userId)
+  override def read(userId: UserId): Option[Play] = {
```

Comment:
you should always return a Future untill you really need the value, in order to avoid blocking situation